backend/lcfs/web/api/allocation_agreement/importer.py

```diff
@@ -0,0 +1,472 @@
+        fuel_type,
+        fuel_type_other,
+        fuel_category,
+        provision,
+        fuel_code,
+        quantity,
+    ) = row
+
+    missing_fields = []
+    if responsibility is None:
+        missing_fields.append("Responsibility")
+    if transaction_partner is None:
+        missing_fields.append("Legal name of transaction partner")
+    if postal_address is None:
+        missing_fields.append("Address for service")
+    if email is None:
+        missing_fields.append("Email")
+    if phone is None:
+        missing_fields.append("Phone")
+    if fuel_type is None:
+        missing_fields.append("Fuel type")
+    if fuel_category is None:
+        missing_fields.append("Fuel category")
+    if provision is None:
+        missing_fields.append("Determining Carbon Intensity")
+    if quantity is None:
+        missing_fields.append("Quantity")
+
+    if missing_fields:
+        return f"Row {row_idx}: Missing required fields: {', '.join(missing_fields)}"
+
+    # Validate email address.
+    email_pattern = re.compile(r"^[\w\.-]+@[\w\.-]+\.\w+$")
+    if not email_pattern.match(email):
+        return f"Row {row_idx}: Invalid email address"
+
+    # Validate quantity (must be integer > 0).
+    try:
+        if int(quantity) <= 0:
+            return f"Row {row_idx}: Quantity must be greater than 0"
+    except Exception:
+        return f"Row {row_idx}: Quantity must be a number"
+
+    # Validate lookups for fuel type, fuel category, and provision.
+    if fuel_type not in valid_fuel_types:
+        return f"Row {row_idx}: Invalid fuel type: {fuel_type}"
+
+    if fuel_category not in valid_fuel_categories:
+        return f"Row {row_idx}: Invalid fuel category: {fuel_category}"
+
+    if provision not in valid_provisions:
+        return f"Row {row_idx}: Invalid determining carbon intensity: {provision}"
+
+    return None
+
+
+def _parse_row(
+    row: tuple, compliance_report_id: int
+) -> AllocationAgreementCreateSchema:
+    """
+    Parses a valid row into an AllocationAgreementCreateSchema object.
+    """
+    (
+        responsibility,
+        transaction_partner,
+        postal_address,
+        email,
+        phone,
+        fuel_type,
+        fuel_type_other,
+        fuel_category,
+        provision,
+        fuel_code,
+        quantity_raw,
+    ) = row
+
+    quantity = int(quantity_raw) if quantity_raw is not None else 0
+
+    return AllocationAgreementCreateSchema(
+        compliance_report_id=compliance_report_id,
+        allocation_transaction_type=responsibility,
+        transaction_partner=transaction_partner,
+        postal_address=postal_address,
+        transaction_partner_email=email,
+        transaction_partner_phone=phone,
+        fuel_type=fuel_type,
+        fuel_type_other=fuel_type_other or None,
+        fuel_category=fuel_category,
+        provision_of_the_act=provision,
+        fuel_code=fuel_code or None,
+        quantity=quantity,
+    )
+
+
+async def _update_progress(
```

Comment:
Future thinking: This looks like a copy from the FSE importer, we should think about adding an JobService that handles the common redis functionality.